graphql/batch_executor.go

```diff
@@ -52,6 +60,7 @@ type UnitResolver func(*WorkUnit) []*WorkUnit
 // work unit.
 type WorkScheduler interface {
 	Run(resolver UnitResolver, startingUnits ...*WorkUnit)
+	RunAll(ctx context.Context, executionUnits []*ExecutionUnit)
```

Comment:
Why does this function take a context, we don't use it.  I think the contexts on the execution Units should be enough.